src/plugins/management/public/components/feature_cards/feature_cards.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+import {
+  EuiCard,
+  EuiFlexGrid,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPageContent,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { AppCategory, ChromeNavLink, CoreStart } from 'opensearch-dashboards/public';
+import React, { useMemo } from 'react';
+
+export interface FeatureCardsProps {
+  pageTitle: string;
+  navLinks: ChromeNavLink[];
+  navigateToApp: CoreStart['application']['navigateToApp'];
+  getStartedCards: Array<{
+    id: string;
+    title: string;
+    description: string;
+  }>;
+}
+
+const getStartedTitle = i18n.translate('management.gettingStarted.label', {
+  defaultMessage: 'Get started',
+});
+
+export const FeatureCards = ({
+  navLinks,
+  navigateToApp,
+  pageTitle,
+  getStartedCards,
+}: FeatureCardsProps) => {
+  const itemsPerRow = 4;
+  const groupedCardForDisplay = useMemo(() => {
+    const grouped: Array<{ category?: AppCategory; navLinks: ChromeNavLink[][] }> = [];
+    navLinks.forEach((link) => {
+      let lastGroup = grouped.length ? grouped[grouped.length - 1] : undefined;
+      if (!lastGroup || lastGroup.category !== link.category) {
```

Comment:
Will the navLinks already sort by category outside? If not the grouped logic may failed.

Reply:
Yeah, let me add some comment here.